src/jacobian.jl

```diff
@@ -144,7 +144,7 @@ reshape_jacobian(result::DiffResult, ydual, xdual) = reshape_jacobian(DiffResult
 ###############
 
 function vector_mode_jacobian(f::F, x, cfg::JacobianConfig{T,V,N}) where {F,T,V,N}
-    ydual = vector_mode_dual_eval!(f, x, cfg)
+    ydual = vector_mode_dual_eval!(cfg, f, x)
```

Comment:
even in `!` functions I think you put the function first to allow for `do` syntax. So it would be `f, cfg, x`.

Reply:
You are really trying to make me regret this...

ok I've changed it again